### PR TITLE
fix(discover2): For count() drilldowns, preserve aggregated fields by converting them to their non-aggregated form

### PR DESCRIPTION
**Pull request by dashed**
For count() drilldowns, preserve aggregated fields by converting them to their non-aggregated form:

- Traditional aggregated columns such as `count_unique(user)` will be converted to `user`.

- Transaction-based columns such as `p75()`, `p95()`, and `p99()` will be converted to `transaction.duration`.

- Transformed columns are de-duplicated. For example, if a table contains `avg(transaction.duration)` `p75()`, `p95()`, and `p99()`, then only `transaction.duration` is added after the transformation.

## TODO

- [x] update tests
- [x] update `last_seen` and others

### CODE REVIEW

File: tests/js/spec/views/eventsV2/utils.spec.jsx [hunk 1]
```diff
@@ -305,12 +305,69 @@ describe('getExpandedResults()', function() {
...
-    const view = new EventView(state);
-    const result = getExpandedResults(view, {}, {});
+  it('preserve aggregated fields', () => {
+    let view = new EventView(state);
+    let result = getExpandedResults(view, {}, {});
 
-    expect(result.fields).toEqual([{field: 'title'}, {field: 'custom_tag'}]);
+    expect(result.fields).toEqual([
+      {field: 'id', width: 300}, // expect count() to be converted to id
+      {field: 'last_seen', width: 300},
```
Comment (markstory) [inline tests/js/spec/views/eventsV2/utils.spec.jsx]: last_seen is an aggregate field on timestamp. It should be replaced.

Comment (dashed) [reply tests/js/spec/views/eventsV2/utils.spec.jsx]: Ah gotcha.

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx [hunk 2]
```diff
@@ -384,23 +384,31 @@ export function downloadAsCsv(tableData, columnOrder, filename) {
   link.remove();
 }
 
+type AggregateTransformer = (field: string) => string;
+
+// an array of 2-tuples of which the first tuple is an array of field names or aggregate names
+// to match a given column on, and the second tuple would transform the given column.
+const TRANSFORM_AGGREGATES: Array<[Array<string>, AggregateTransformer]> = [
+  [['p99', 'p95', 'p75'], () => 'transaction.duration'],
```
Comment (evanh) [inline src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: What about apdex and impact?

**Review by evanh — APPROVED** [review]

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx (hunk 2 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: `apdex` and `impact` will now be dropped.

File: src/sentry/static/sentry/app/views/eventsV2/utils.tsx [hunk 3]
```diff
@@ -423,10 +436,83 @@ export function getExpandedResults(
...
+  const fieldsToDelete: number[] = [];
+
+  // make a best effort to transform aggregated columns with its non-aggregated form
+  fieldsToUpdate.forEach((indexToUpdate: number) => {
+    const currentField: FieldType = nextView.fields[indexToUpdate];
+    const exploded = explodeField(currentField);
+
+    // check if we can use an aggregated transform function
+
+    for (const tuple of TRANSFORM_AGGREGATES) {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/utils.tsx]: If TRANSFORM_AGGREGATES was a lookup of fields => transforms you could skip a layer of loops. :golf:

**Review by markstory — APPROVED** [review]

**Pull request merged**